packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -64,84 +67,82 @@ describe('Model > SubjectStore', function () {
       })
 
       it('should request more subjects', function () {
-        expect(populateSpy).to.have.been.calledOnce()
+        // Once for initialization and once after the queue has been advanced to less than 3 subjects
+        expect(rootStore.subjects.populateQueue).to.have.been.calledTwice()
+      })
+
+      describe('when the initial response has less than three subjects', function () {
+        let rootStore
+        before(function () {
+          const clientStub = stubPanoptesJs({ workflows: workflow, subjects: shortListSubjects })
+          rootStore = setupStores(clientStub)
+        })
+
+        it('should request more subjects', function () {
+          // Once for initialization and again since less than three subjects in initial response
+          expect(rootStore.subjects.populateQueue).to.have.been.calledTwice()
+        })
+      })
+
+      describe('when the initial response has no subjects', function () {
+        let rootStore
+        before(function () {
+          const clientStub = stubPanoptesJs({ workflows: workflow, subjects: [] })
+          rootStore = setupStores(clientStub)
+        })
+
+        it('should request more subjects', function () {
+          // Once for initialization
+          expect(rootStore.subjects.populateQueue).to.have.been.calledOnce()
+        })
       })
     })
 
     describe('after emptying the queue', function () {
+      let rootStore
       before(function () {
-        sinon.stub(clientStub.panoptes, 'get').callsFake(() => Promise.resolve({ body: [] }))
+        const clientStub = stubPanoptesJs({ workflows: workflow, subjects: [] })
+        rootStore = setupStores(clientStub)
+      })
+
+      beforeEach(function () {
         while (rootStore.subjects.resources.size > 0) {
           rootStore.subjects.advance()
         }
       })
 
-      after(function () {
-        clientStub.panoptes.get.restore()
-      })
-
       it('should leave the active subject empty', function () {
```

Comment:
```suggestion
      it('should advance the queue until it is empty', function () {
```
I think this is also a better description of the test. I'm thinking maybe expect `rootStore.subjects.advance` to have been called here?

Reply:
I don't think that's what we're testing here. We're testing to make sure active isn't being set when the resources size is 0. I think if we want to test advance is being called as its written in the store, we'll need integration tests testing the observers setup between the classification store and the subjects store. If we add an expectation that advance is called here, we're just testing the calls from the `beforeEach` which doesn't seem right to me.

Reply:
Sorry but II wrote this test and this is what it was intended to test. It should start with a populated queue, advance that queue until it’s empty then check that it really is empty. It’s a regression test to catch a bug that happened if you call advance on the queue when the number of subjects is small.

Reply:
Ok, we can include the suggested refactor here in the issue about improving these tests.